app/scripts/controllers/network/network.js

```diff
@@ -155,6 +155,11 @@ module.exports = class NetworkController extends EventEmitter {
   }
 
   _configureStandardProvider ({ rpcUrl }) {
+    const colon = rpcUrl.indexOf(':')
+    if (colon !== -1) {
+      const proto = rpcUrl.substring(0, colon)
```

Comment:
Instead of using indexOf and substring here, can we use the URL module to pick the scheme out of the URL. We should be able to create a [`URL`](https://nodejs.org/docs/latest-v8.x/api/url.html#url_constructor_new_url_input_base) instance and mutate the protocol.